.github/workflows/spark_master_test.yaml

```diff
@@ -42,11 +42,52 @@ jobs:
           sudo apt-get update
           sudo apt-get install -y make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
           sudo apt install libedit-dev
-        if: steps.git-diff.outputs.diff
+          curl -LO https://github.com/bufbuild/buf/releases/download/v1.28.1/buf-Linux-x86_64.tar.gz
+          mkdir -p ~/buf
+          tar -xvzf buf-Linux-x86_64.tar.gz -C ~/buf --strip-components 1
+          rm buf-Linux-x86_64.tar.gz
+          sudo apt install python3-pip --fix-missing
+          sudo pip3 install pipenv==2021.5.29
+          curl https://pyenv.run | bash
+          export PATH="~/.pyenv/bin:$PATH"
+          eval "$(pyenv init -)"
+          eval "$(pyenv virtualenv-init -)"
+          pyenv install 3.9
+          pyenv global system 3.9
+          pipenv --python 3.9 install
+          # Update the pip version to 24.0. By default `pyenv.run` installs the latest pip version
+          # available. From version 24.1, `pip` doesn't allow installing python packages
+          # with version string containing `-`. In Delta-Spark case, the pypi package generated has
+          # `-SNAPSHOT` in version (e.g. `3.3.0-SNAPSHOT`) as the version is picked up from
+          # the`version.sbt` file.
+          pipenv run pip install pip==24.0 setuptools==69.5.1 wheel==0.43.0
+          pipenv run pip install flake8==3.9.0
+          pipenv run pip install black==23.9.1
+          pipenv run pip install importlib_metadata==3.10.0
+          pipenv run pip install mypy==1.8.0
+          pipenv run pip install mypy-protobuf==3.3.0
+          pipenv run pip install cryptography==37.0.4
+          pipenv run pip install twine==4.0.1
+          pipenv run pip install wheel==0.33.4
+          pipenv run pip install setuptools==41.1.0
+          pipenv run pip install pydocstyle==3.0.0
+          pipenv run pip install pandas==2.0.0
+          pipenv run pip install pyarrow==10.0.0
+          pipenv run pip install pypandoc==1.3.3
+          pipenv run pip install numpy==1.21
+          pipenv run pip install grpcio==1.62.0
+          pipenv run pip install grpcio-status==1.62.0
+          pipenv run pip install googleapis-common-protos==1.56.4
+          pipenv run pip install protobuf==4.25.1
+          pipenv run pip install googleapis-common-protos-stubs==2.2.0
+          pipenv run pip install grpc-stubs==1.24.11
+          pipenv run pip install pyspark==4.0.0.dev1
       - name: Run Spark Master tests
         # when changing TEST_PARALLELISM_COUNT make sure to also change it in spark_test.yaml
         run: |
+          echo 'ThisBuild / version := "4.0.0-SNAPSHOT"' > version.sbt     
```

Comment:
This is used in python code to determine scala/python versions, whether to include delta connect packages etc

Reply:
so this overrides the existing `version.sbt`? would this show up as a non-empty git diff, meaning that the tests are always triggered?

couldn't we do this inside of `build.sbt` like how we create the Meta class (which includes the release version)

Reply:
I don't think this will be an issue for the git diff since the diff check is executed _before_ this step